File: Algorithms/299. Bulls and Cows/Solution.py
#coding=utf-8
__author__ = 'xuxuan'
class Solution(object):
    def getHint(self, secret, guess):
        """
        :type secret: str
        :type guess: str
        :rtype: str
        """
        s=list(secret)
        g=list(guess)
        a=0
        b=0
        i=0
        while i<len(s):
            if s[i]==g[i]:
                a+=1
                s.remove(s[i])
                g.remove(g[i])
            else:
                i+=1
        i=0
        while i<len(g):
            if g[i] in s:
                s.remove(g[i])
                g.remove(g[i])
                b+=1
            else:
                i+=1
        return str(a)+'A'+str(b)+'B'

File: Algorithms/299. Bulls and Cows/Solution.java
public class Solution {
    public String getHint(String secret, String guess) {
        int length=secret.length();
        if (length==0) {
            return "0A0B";
        }
        char[] cs=secret.toCharArray();
        char[] cg=guess.toCharArray();
        boolean[] f=new boolean[length];
        int a=0;
        for (int i = 0; i < length; i++) {
            boolean flag=cs[i]==cg[i];
            f[i]=flag;
            a+=flag?1:0;
        }
        boolean[] g=new boolean[length];
        int b=0;
        for (int i = 0; i < length; i++) {
            for (int j = 0; j < length; j++) {
                if (i!=j&&!g[j]&&!f[i]&&!f[j]&&cg[i]==cs[j]) {
                    b++;
                    g[j]=true;
                    break;
                }
            }
        }
        return a+"A"+b+"B";
    }
}